gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py

```diff
@@ -60,63 +61,140 @@ def __init__(self, info_keys_to_move_to_calls_regex, copy_quality_to_calls,
     start = variant.start - (variant.start % self._window_size)
     for i in range(start, variant.end, self._window_size):
       yield ':'.join([str(x) for x in [variant.reference_name, i]])
 
+  def get_merged_variants(self, variants, key=None):
+    non_variant_tree = IntervalTree()
+    grouped_variants = collections.defaultdict(list)
+    for v in variants:
+      self._align_with_window(v, key)
+      if self._is_non_variant(v):
+        non_variant_tree.addi(v.start, v.end, v)
+      else:
+        group_key = next(self._move_to_calls.get_merge_keys(v))
+        grouped_variants[group_key].append(v)
+
+    non_variants = self._merge_non_variants(non_variant_tree)
+    variants = self._merge_variants(grouped_variants)
+
+    non_variant_tree.clear()
+    for nv in non_variants:
+      non_variant_tree.addi(nv.start, nv.end, nv)
+
+    splits = IntervalTree()
+    for v in variants:
+      non_variant_interval = non_variant_tree.search(v.start, v.end)
+      if non_variant_interval:
+        non_variant = next(iter(non_variant_interval)).data
+        v.calls.extend(non_variant.calls)
+        v.calls = sorted(v.calls)
+        self._update_splits(splits, v)
+      yield v
+
+    for non_variant in self._split_non_variants(non_variant_tree, splits):
+      yield non_variant
+
+  def _merge_non_variants(self, t):
+    bounds = sorted(set([p for v in t for p in [v.begin, v.end]]))
+    intervals = [(bounds[i], bounds[i+1]) for i, _ in enumerate(bounds[:-1])]
+    merged_non_variants = []
+    for start, end in intervals:
+      overlapping_non_variants = t[start:end]
+      if not overlapping_non_variants:
+        continue
+
+      overlapping_variants = [ov.data for ov in overlapping_non_variants]
+      merged_non_variant = copy.deepcopy(overlapping_variants[0])
+      merged_non_variant.reference_bases = None
+      merged_non_variant.start = start
+      merged_non_variant.end = end
+
+      for non_variant in overlapping_variants[1:]:
+        merged_non_variant.names.extend(non_variant.names)
+        merged_non_variant.filters.extend(non_variant.filters)
+        merged_non_variant.quality = min(merged_non_variant.quality,
+                                         non_variant.quality)
+        merged_non_variant.calls.extend(non_variant.calls)
+
+      merged_non_variant.names = sorted(set(merged_non_variant.names))
+      merged_non_variant.filters = sorted(set(merged_non_variant.filters))
+      merged_non_variant.calls = sorted(merged_non_variant.calls)
+
+      merged_non_variants.append(merged_non_variant)
+
+    return merged_non_variants
+
+  def _merge_variants(self, grouped_variants):
+    merged_variants = []
+    for merge_key, variants in grouped_variants.iteritems():
+      merged_variants.extend(
+          self._move_to_calls.get_merged_variants(variants, merge_key))
+    return merged_variants
+
+  def _update_splits(self, splits, v):
+    # Increase split size to also combine adjacent splits
```

Comment:
nit: please put this comment just above overlapping_splits.

Reply:
Whoops, looks like I moved some other stuff around and forgot this comment. Good catch!